src/map/packet_system.cpp

```diff
@@ -3183,16 +3189,37 @@ void SmallPacket0x04E(map_session_data_t* const PSession, CCharEntity* const PCh
                     return;
                 }
 
-                const char* fmtQuery = "INSERT INTO auction_house(itemid, stack, seller, seller_name, date, price) VALUES(%u,%u,%u,'%s',%u,%u)";
+                // Order here matters. We first make sure we CAN remove the item from the player's inventory,
+                // and then we put the item up for sale. Otherwise the item could go on sale without it being removed
+                // from the player's inventory.
+
+                // 1. Check we can afford AH fee. This needs to be done before updating item so we are sure
+                // we can perform the whole operation.
+                if (!charutils::ValidateUpdateItem(PChar, LOC_INVENTORY, 0, -(int32)auctionFee)) {
```

Comment:
This is kinda redundant, but we must check that player can afford AH fee BEFORE actually removing the item from inventory. Currently we do both regardless of their result.